vendor/assets/javascripts/browser-update.js.erb

```diff
@@ -21,10 +21,14 @@ var $buo = function() {
   document.getElementsByTagName('body')[0].classList.add("crawler");
   var mainElement = document.getElementById("main");
   var noscriptElements = document.getElementsByTagName("noscript");
-  // noscriptElements[0].innerHTML contains encoded HTML
-  var innerHTML = noscriptElements[0].childNodes[0].nodeValue;
-  mainElement.innerHTML = innerHTML;
-
+  // find the element with the "data-path" attribute set
+  for (var i = 0; i < noscriptElements.length; ++i) {
+    if (noscriptElements[i].dataset["path"]) {
+      // noscriptElements[i].innerHTML contains encoded HTML
```

Comment:
I know you didn't touch that part of the code but why does it contain encoded HTML?

Reply:
The [noscript](https://html.spec.whatwg.org/multipage/scripting.html#the-noscript-element) element must contain only text and it is treated as such.

Reply:
Are you sure about that? That's not what I understand/read and what I've experienced.

Reply:
It is true in IE and `document.getElementsByTagName("noscript")[0].innerHTML` is always encoded.